cpp/src/arrow/array/array_nested.h

```diff
@@ -43,30 +43,31 @@ namespace arrow {
 /// @{
 
 // ----------------------------------------------------------------------
-// ListArray
+// VarLengthListLikeArray
 
 template <typename TYPE>
-class BaseListArray;
+class VarLengthListLikeArray;
```

Comment:
Keep a `Base` prefix here as well?

Reply:
Waiting for an answer of my question on the thread above.